app/screens/Onboarding/Joint/Loading/LoadingScreen.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { SafeAreaView } from 'react-native';
+import { useTranslation } from 'react-i18next';
+import _ from 'lodash';
+
+import { BottomButtonView, ExtendedText, Timer } from '../../../../components';
+import { generalStyles } from '../../../../utils/styles';
+import { ILoadingScreenProps } from './LoadingScreen.props';
+import { SPEECH_LIST } from './LoadingScreen.data';
+
+export const LoadingScreen: React.FC<ILoadingScreenProps> = () => {
+  const { t } = useTranslation();
+
+  const MAX_TIME = 100;
+  const INTERVAL_INCREASE = 1;
+
+  const [currentSpeechIdx, setCurrentSpeechIdx] = useState(0);
+  const [currentLoaderValue, setCurrentLoaderValue] = useState(0);
+
+  const goToNextSpeech = useCallback(() => {
+    setCurrentSpeechIdx(currentSpeechIdx + 1);
+  }, [currentSpeechIdx]);
+
+  const getCurrentSpeech = useCallback(() => {
+    return SPEECH_LIST[currentSpeechIdx];
+  }, [currentSpeechIdx]);
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setCurrentLoaderValue(currentLoaderValue + INTERVAL_INCREASE);
+      if (currentLoaderValue === 50) {
+        goToNextSpeech();
+      }
+    }, 100);
+
+    if (currentLoaderValue >= MAX_TIME) {
+      clearInterval(interval);
+    }
+
+    return () => clearInterval(interval);
+  }, [currentLoaderValue, goToNextSpeech]);
```

Comment:
1. Please, make the constant for `50` magic number, because this value should be changed after the time.
2. Please, make the constant for `100` tick value (also magic number anti-pattern), because this value should be changed after the time from the customer side.